test/gtest/ucs/test_profile.cc

```diff
@@ -164,14 +165,22 @@ void test_profile::run_profiled_code(int num_iters)
     if (num_threads() == 1) {
         profile_thread_func(&param);
     } else {
-        pthread_t threads[num_threads()];
+        std::queue<pthread_t> threads;
+
         for (int i = 0; i < num_threads(); ++i) {
-            pthread_create(&threads[i], NULL, profile_thread_func,
-                           (void*)&param);
+            pthread_t profile_thread;
+            ret = pthread_create(&profile_thread, NULL, profile_thread_func,
+                                 (void*)&param);
+            ASSERT_EQ(0, ret);
```

Comment:
there is a problem with error handling: previously created threads are not cleaned

Reply:
done